src/test/java/com/squareup/javapoet/JavaFileTest.java

```diff
@@ -25,6 +26,141 @@
 
 @RunWith(JUnit4.class)
 public final class JavaFileTest {
+  @Test(expected = IllegalStateException.class)
+  public void importStaticFailsForChainedTypeName() {
```

Comment:
Keep this state check "as is", or should I work around this strange case of `TypeName` chaining? Maybe like, emit the old pendine one and remember the current for the next round?